sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubProducerAsyncClient.java

```diff
@@ -132,6 +141,25 @@
         this.retryPolicy = getRetryPolicy(retryOptions);
         this.scheduler = scheduler;
         this.isSharedConnection = isSharedConnection;
+        this.isIdempotentPartitionPublishing = isIdempotentPartitionPublishing;
+        if (isIdempotentPartitionPublishing) {
+            if (initialPartitionPublishingStates == null) {
+                this.partitionPublishingStates = new HashMap<>();
+            } else {
+                this.partitionPublishingStates = initialPartitionPublishingStates;
+            }
+        } else {
+            this.partitionPublishingStates = null;
+        }
+    }
+
+    EventHubProducerAsyncClient(String fullyQualifiedNamespace, String eventHubName,
+        EventHubConnectionProcessor connectionProcessor, AmqpRetryOptions retryOptions, TracerProvider tracerProvider,
```

Comment:
Same with using options for the constructor.

Reply:
Same as above